api/classInstances/classInstanceMiddleware.js

```diff
@@ -8,14 +9,13 @@ const { findByInstructorId } = require('../instructor/instructorModel');
 const { getAllCourseTypes } = require('../courseTypes/courseTypesModel');
 
 const checkClassInstanceExist = async (req, res, next) => {
-  const class_id = req.params.class_id;
+  const class_id = parseInt(req.params.class_id);
```

Comment:
Another technique you could explore here is `Number()` or `String()` or even `.toString()`